File: backend/App.DAL.DTO/Firm.cs
using Base.Domain;

namespace App.DAL.DTO;

public class Firm : BaseEntityId
{
    public string Name { get; set; } = default!;
    public string RegistryCode { get; set; } = default!;
    public int ParticipantCount { get; set; }
    public string AdditionalNotes { get; set; } = default!;
    

    
    public List<ParticipantEvent>? ParticipantEvents { get; set; }
}

File: frontend/entities/Firm.ts
import ParticipantEvent from "./ParticipantEvent";

export default class Firm {
    id: string;
    name: string;
    registryCode: string;
    participantCount: number;
    participantEvents: ParticipantEvent[];

    constructor(
        id: string,
        name: string,
        registryCode: string,
        participantCount: number,
        participantEvents: ParticipantEvent[] 
    ) {
        this.id = id;
        this.name = name;
        this.registryCode = registryCode;
        this.participantCount = participantCount;
        this.participantEvents = participantEvents 
    }
}
